README.md

```diff
@@ -1 +1,164 @@
+class ExampleOpensearchIndexStack(Stack):
+    def __init__(self, scope: Stack, opensearch_domain_endpoint: str) -> None:
+        super().__init__(scope=scope, id='ExampleOpensearchIndexStack')
+
+        # Define index settings.
+        index_settings = {
+             'index.refresh_interval': '2s',
+        }
+
+        # Define index mappings.
+        index_mappings = {
+            'id': {
+                'type': 'keyword'
+            },
+            'author': {
+                'type': 'keyword'
+            },
+            'title': {
+                'type': 'keyword'
+            },
+            'content': {
+                'type': 'text'
+            },
+            'timestamp': {
+                'type': 'date',
+                'format': 'yyyy-MM-dd HH:mm:ss||epoch_millis'
+            },
+        }
+
+        # Initialize AWS CloudFormation custom resource using given OpenSearch
+        # domain endpoint with defined settings and mappings of the index.
+        self.index = OpensearchIndex(
+            scope=self,
+            name='OpensearchIndex',
+            opensearch_domain_endpoint=opensearch_domain_endpoint,
+            index_prefix='example-index-prefix',
+            index_settings=index_settings
+            index_mapping_settings=index_mappings
+        )
+
+    @property
+    def index_name(self) -> str:
+        """
+        Retrieve OpenSearch index name.
+
+        :return: String representation of index name.
+        """
+
+        return self.index.index_name
+
+```
+
+In the above example, we created two dictionaries holding settings and mappings of the index. More information about settings and mappings of the index can be found here https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/.
+
+Then initialized custom resource using OpenSearch domain endpoint together with prefix, settings, and mappings of the index.
+
+Index prefix can be any string that complies with index naming restrictions listed in section **Known limits**.
+
+In the case of index name (index prefix) change, all documents residing (indexed using that index) on OpenSearch will be reindexed using a new index name. If there is a need to skip reindex of the documents **`reindex`** flag must be set explicitly with the value **`False`**.
+
+```python
+
+self.index = OpensearchIndex(
+    scope=self,
+    name='OpensearchIndex',
+    opensearch_domain_endpoint=opensearch_domain_endpoint,
+    index_prefix='example-index-prefix',
+    index_settings=index_settings
+    index_mapping_settings=index_mappings,
+    reindex=False
+)
+```
+
+Following parameters are optional and can be omitted:
+
+- `index_settings`
+- `index_mappings`
+- `reindex`
+
+### Known limits
+
+There are some limitations. Currently, reindex of documents is not possible in the case of the OpenSearch domain endpoint change. This package will create a new index at the new endpoint and log a warning of unsuccessful reindex of the documents.<br><br>
```

Comment:
Currently, reindex of documents is not possible if the OpenSearch domain endpoint changes